biotech-agentic-analyst/datalab_pipeline/pipeline.py

```
@@ -0,0 +1,289 @@
+
+def _poll_for_scores(check_url: str, api_key: str, max_wait: int = 60) -> float | None:
+    """Keep polling request_check_url until extraction_score_average appears."""
+    headers = {"X-API-Key": api_key}
+    deadline = time.time() + max_wait
+    while time.time() < deadline:
+        try:
+            data = _http.get(check_url, headers=headers, timeout=10).json()
+            if "extraction_score_average" in data:
+                return float(data["extraction_score_average"])
+        except Exception:
+            pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)